### ISSUE TITLE
Request to list all missing crates rather than first one

### ISSUE THREAD

**Issue by gilescope** [rust-lang/cargo#8632]
Request to list all missing crates rather than exit on the first one.

A bespoke but not entirely unlikely use case:

 As a company that has locally cached crates in a dir via some bespoke process. when the source code is updated and depends on different crate versions we run cargo check and get a the first missing crate/version. Then we repackage (somehow) to include that crate version, then we repeat until we have them all.

It would be nice if cargo listed all the missing crates/versions it needed but couldn't find before it exited.

**Comment by Eh2406** [comment]
I am not entirely sure I know which error you are referring to. Do you mean the case of
```
[dependencies]
not-a-thing="*"
not-another-thing="*"
```
whos error only menchens one of them?

In general listing all the transitive deps that are not available is hard. Even just proving that there is no way around any one problem is np-complete. So listing all of them may be a lot of work. But, it seems reasonable to add a bespoke pass over the direct dependencies, before resolution. Is it is worth the code complexity? Depends on implementation details, that I can't remember at the moment.

**Comment by gilescope** [comment]
Yes that's right. If it could tell me all the immediate dependencies it's missing that would be good. At the moment, it's one by one that I'm filling in blanks. If it could tell me one level at a time that would significantly cut down the number of runs I need to do to find out the information.

This is an example error:
```
error: failed to select a version for the requirement `chrono-tz = "=0.5.2"`
  candidate versions found which didn't match: 0.5.3
```
